platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/capabilities/CapabilitiesLocalComponentIntegrationTest.groovy

```diff
@@ -76,5 +77,64 @@ class CapabilitiesLocalComponentIntegrationTest extends AbstractIntegrationSpec
         then:
         outputContains("Could not resolve project :.")
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/26377")
+    def "ResolvedVariantResults reported by ResolutionResult and ArtifactCollection have same capabilities when they are added to configuration in hierarchy"() {
+        settingsFile << "include 'producer'"
+        file("producer/build.gradle") << """
+            group="com.foo"
+
+            task zip(type: Zip) {
+                archiveFileName = "producer.zip"
+                destinationDirectory = layout.buildDirectory
+            }
+
+            configurations {
+                dependencyScope("api") {
+                    outgoing {
+                        capability("com.foo:producer:2.0")
+                        capability("org.bar:extra-capability:1.0")
+                    }
+                }
+                consumable("elements") {
+                    extendsFrom(api)
+                    outgoing.artifact(tasks.zip)
+                    attributes {
+                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.class, Category.LIBRARY))
+                    }
+                }
+            }
+        """
+
+        buildFile << """
+            configurations {
+                dependencyScope("implementation")
+                resolvable("classpath") {
+                    extendsFrom(implementation)
+                    attributes {
+                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.class, Category.LIBRARY))
+                    }
+                }
+            }
+
+            dependencies {
+                implementation(project(":producer"))
+            }
+
+            task resolve {
+                def conf = configurations.classpath
+                def root = conf.incoming.resolutionResult.root
+                def artifactVariants = conf.incoming.artifacts.artifacts.collect { it.variant }
+                doLast {
+                    def graphVariant = root.dependencies.find { it.selected.id.projectPath == ":producer" }.resolvedVariant
+                    def artifactVariant = artifactVariants.find { it.owner.projectPath == ":producer" }
+                    assert graphVariant.capabilities == artifactVariant.capabilities
```

Comment:
❌ can you also assert what the capabilities are here?  That makes it a little more clear which capabilities are supposed to be here because it would also be valid for both capabilities sets to be empty.